app/components/Tooltip/index.js

```diff
@@ -158,55 +111,292 @@ function makeBody(content, currentLocal) {
       const closestFreeSlot = getClosestAvailableSlot(data);
       const nextReservedSlot = getNextReservedSlot(data);
 
-      return (
-        <>
-          <Title>{translate(name)}</Title>
-          {maxPeopleCount && (
-            <Row>
+      return {
+        type: SpaceContentTypes.SINGLE,
+        name,
+        vacancyStatus,
+        maxPeopleCount,
+        description,
+        closestFreeSlot,
+        nextReservedSlot,
+      };
+    }
+    case SpaceContentTypes.GROUP: {
+      const { category } = spaces.find(space => space.category);
+      const spacesByType = Object.entries(groupBy(spaces, 'type'));
+      const groups = spacesByType.map(([spaceType, spacesWithType]) => {
+        const totalCount = spacesWithType.length;
+        const availableCount = getAvailableCount(spacesWithType);
+
+        return {
+          id: spaceType,
+          availableCount,
+          totalCount,
+          spaceType,
+        };
+      });
+
+      return {
+        type: SpaceContentTypes.GROUP,
+        category,
+        groups,
+      };
+    }
+    case SpaceContentTypes.GRID: {
+      const spacesByType = Object.entries(groupBy(spaces, 'type'));
+
+      if (spacesByType.length === 0 || spacesByType > 1) {
+        throw Error(
+          'The GRID SpaceContentType supports rendering of exactly one SpaceType group',
+        );
+      }
+
+      // SpacesByType should be exactly 1 in length to we an safely only
+      // take the first item into consideration.
```

Comment:
Im not sure what you are trying to say here :sweat_smile:

Reply:
Good catch :D I'll fix it.